runtime/compiler/trj9/control/CompilationThread.cpp

```diff
@@ -6403,97 +6377,41 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
-            }
-
-         if (protectedResult == 0) // successful completion of wrappedCompile
-            metaData = reinterpret_cast<TR_MethodMetaData *>(result);
-         else
-            metaData = 0;
-         }
-      else
-#endif
-         metaData = wrappedCompile(privatePortLibrary, &compParam);
-      }
-   else
-      {
-      entry->_compErrCode = compilationRestrictedMethod;
-      // can't compile this method: mark it appropriately
-      //if (!entry->isDLTCompile())
-      //   startPC = compilationEnd(vmThread, jitConfig, method,
-      //                            0, entry->_oldStartPC, entry->_classForNewInstance, vm);
-      }
-
-   if (vmThread)
-      {
-      vmThread->omrVMThread->vmState = oldState;
-      }
-
-
+   return vm;
+   }
+
+/**
+ * @brief TR::CompilationInfoPerThreadBase::postCompilationTasks
+ * @param vmThread Pointer to the current J9VMThread
+ * @param entry Pointer to the method to be compiled entry
+ * @param method Pointer to the J9Method
+ * @param aotCachedMethod Pointer to the AOT code to be relocated (if it exists)
+ * @param metaData Pointer to the TR_MethodMetaData created on a successful compile
+ * @param compParam Reference to the CompileParameters associated with the current compilation
+ * @param canDoRelocatableCompile Indicates if the current compilation could be AOT compiled
+ * @param eligibleForRelocatableCompile Indicates if the current compilation was eligible for AOT
+ * @param reloRuntime Pointer to a TR_RelocationRuntime; NULL if JVM does not have AOT support
+ * @param scratchMemoryAvailable Indicates if the scratch memory infrastructre for the current compilation can be used
+ * @return StartPC of the compiled body, NULL if the compilation failed
+ *
+ * This method is used to perform a set of tasks needed after attempting a compilation. It is important to note
+ * that the scratch memory infrastructure (TR_Memory, TR::Region, TR::SegmentAllocator) may not exist within this
+ * method as this method is called in both the try and catch block in TR::CompilationInfoPerThreadBase::compile.
+ */
+void *
+TR::CompilationInfoPerThreadBase::postCompilationTasks(J9VMThread * vmThread,
+                                                       TR_MethodToBeCompiled *entry,
+                                                       J9Method *method,
+                                                       const void *aotCachedMethod,
+                                                       TR_MethodMetaData *metaData,
+                                                       CompileParameters &compParam,
+                                                       bool canDoRelocatableCompile,
+                                                       bool eligibleForRelocatableCompile,
+                                                       TR_RelocationRuntime *reloRuntime,
+                                                       bool scratchMemoryAvailable)
```

Comment:
I saw that `scratchMemoryAvailable` is not used in this function. Where or how could it be used?

Reply:
I intended this as a means of determining if one can use the `TR_Memory` and `TR::Region` associated with the comp object to do any sort of temporary memory allocation. However, I realize that a non-NULL `comp` object implies that the the scratch memory infrastructure is available so this bool is unnecessary. Additionally, no one should be using any scratch memory since it might result in a `std::bad_alloc` thrown and, as stated in the previous comment, an exception should not escape `postCompilationTasks`.